api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -1,18 +1,242 @@
+  principal_id: participantId,
+  highest_state: 'Graded',
+  most_recent_submitted_date: '2023-02-09 13:23:45',
+  total_num_submissions: 1,
+  highest_score: 10,
+};
+
+const facilitatorAssessmentSubmissionsSummary: FacilitatorAssessmentSubmissionsSummary =
+  {
+    num_participants_with_submissions: 6,
+    num_program_participants: 12,
+    num_ungraded_submissions: 2,
+  };
+
+const emptyAssessmentsSummaryList: AssessmentSummary[] = [];
 
 describe('assessmentsRouter', () => {
   const appAgent = createAppAgentForRouter(assessmentsRouter);
   const exampleAssessmentId = 1;
   const exampleSubmissionId = 1;
 
   describe('GET /', () => {
-    it('should respond with a list of all assessments', done => {
-      const response = { behaviour: 'Shows a list of all assessments' };
-      appAgent.get('/').expect(200, itemEnvelope(response), err => {
-        done(err);
-      });
+    const unenrolledPrincipalId = 4;
+    const enrolledParticipantPrincipalId = 5;
+    const facilitatorPrincipalId = 6;
+    const programIds: number[] = [];
+
+    it('should respond with an empty list for a user not enrolled in any programs', done => {
+      // mock response from function that gets a list of programs the user is enrolled in to return an empty list
+      mockPrincipalEnrolledPrograms.mockResolvedValue(programIds);
+
+      appAgent
+        .get('/')
+        .expect(
+          200,
+          collectionEnvelope(emptyAssessmentsSummaryList, 0),
+          err => {
+            // call a (mock) function that gets a list of programs the user is enrolled in
+            expect(mockPrincipalEnrolledPrograms).toHaveBeenCalledWith(
+              unenrolledPrincipalId
+            );
+            done(err);
+          }
+        );
+    });
+
+    it('should respond with a list of all assessments (without questions) for participant enrolled in one program', done => {
+      mockPrincipalId(enrolledParticipantPrincipalId);
+      const participantAssessmentListResponse = [
+        {
+          curriculum_assessment: curriculumAssessment,
+          program_assessment: programAssessment,
+          submissions_summary: assessmentSubmissionsSummary,
+        },
+      ];
+      // mock response from (function that gets a list of programs the user is enrolled in) to include one program
+      mockPrincipalEnrolledPrograms.mockResolvedValue([4]);
+      // mock response from (call a function that returns the permission of the user for each program (participant/facilitator)) to respond with participant for that one program
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([
+        exampleAssessmentId,
+      ]);
+      mockFindRoleInProgram.mockResolvedValue({ title: 'participant' });
+      // mock response from (get a list of program assessments for each program the user is enrolled in) to respond with a list of program assessments
+      mockGetAssessmentsForProgram.mockResolvedValue(programAssessment);
```

Comment:
I would imagine this function would return an array of objects rather than a singular object, no?

Reply:
Hi, quick question. To be able to use an array here like this `mockGetAssessmentsForProgram.mockResolvedValue([programAssessment])`, I need to change the promise in the `getAssessmentsForProgram` function.  It returns just an object. So, I am not sure that I am able to resolve this suggestion without changing `assessmentService.ts` file.

Reply:
Understood; we'll definitely need to change that function in the next ticket. For now, I'm approving you to change this to `mockGetAssessmentsForProgram.mockResolvedValue([programAssessment])`, even if it gives you an error in VSCode.